packages/passport-interface/src/CredentialManager.ts

```diff
@@ -0,0 +1,145 @@
+import { PCDCollection } from "@pcd/pcd-collection";
+import { ArgumentTypeName, SerializedPCD } from "@pcd/pcd-types";
+import { SemaphoreIdentityPCDPackage } from "@pcd/semaphore-identity-pcd";
+import { SemaphoreSignaturePCDPackage } from "@pcd/semaphore-signature-pcd";
+import { Identity } from "@semaphore-protocol/identity";
+import {
+  FeedCredentialPayload,
+  createFeedCredentialPayload
+} from "./FeedCredential";
+import { CredentialRequest } from "./SubscriptionManager";
+
+export interface CredentialManagerAPI {
+  canGenerateCredential(req: CredentialRequest): boolean;
+  requestCredential(req: CredentialRequest): Promise<SerializedPCD>;
+}
+
+const CREDENTIAL_CACHE_MAX_AGE = 1000 * 60 * 60;
+
+interface CredentialCacheEntry {
+  value: SerializedPCD;
+  timestamp: number;
+}
+
+/**
+ * Handles generation of credentials for feeds.
+ */
+export class CredentialManager implements CredentialManagerAPI {
+  private readonly identity: Identity;
+  private readonly pcds: PCDCollection;
+  private readonly cache: Map<string, CredentialCacheEntry>;
+
+  public constructor(identity: Identity, pcds: PCDCollection) {
+    this.identity = identity;
+    this.pcds = pcds;
+    this.cache = new Map();
```

Comment:
it would be nice if this cache persisted across restarts of the tab, or at least if we were using the same instance of `CredentialManager` across the entire application. I see you're instantiating `CredentialManager` wherever it's needed rather than keeping one instance alive throughout the lifetime of the application.

Reply:
Yes. This only solves the really simple problem of re-using the credential when polling multiple subscriptions at the same time.

For this to be longer-lived, we would need to figure out how it gets updates when the PCD collection changes, which _might_ be trivially easy but would also be a source of weird bugs if it were done incorrectly.